nx-X11/programs/Xserver/hw/nxagent/NXmiwindow.c

```diff
@@ -1,1209 +0,0 @@
-	if (anyMarked && pScreen->PostValidateTree)
-	    (*pScreen->PostValidateTree)(pLayerWin->parent, pFirstChange,
-					  VTOther);
-    }
-    else if (bsExposed)
-    {
-	(*pScreen->WindowExposures) (pWin, NullRegion, bsExposed);
-	RegionDestroy(bsExposed);
-    }
-    if (pWin->realized)
-	WindowsRestructured ();
-}
-
-WindowPtr
-miGetLayerWindow(pWin)
-    WindowPtr pWin;
-{
-    return pWin->firstChild;
-}
-
-#ifdef SHAPE
-/******
- *
- * miSetShape
- *    The border/window shape has changed.  Recompute winSize/borderSize
- *    and send appropriate exposure events
- */
-
-void
-miSetShape(pWin)
-    register WindowPtr	pWin;
-{
-    Bool	WasViewable = (Bool)(pWin->viewable);
-    register ScreenPtr pScreen = pWin->drawable.pScreen;
-    Bool	anyMarked = FALSE;
-    RegionPtr	pOldClip = NULL, bsExposed;
-#ifdef DO_SAVE_UNDERS
-    Bool	dosave = FALSE;
-#endif
-    WindowPtr   pLayerWin;
-
-    if (WasViewable)
-    {
-	anyMarked = (*pScreen->MarkOverlappedWindows)(pWin, pWin,
-						      &pLayerWin);
-	if (pWin->valdata)
-	{
-	    if (HasBorder (pWin))
-	    {
-		RegionPtr	borderVisible;
-
-		borderVisible = RegionCreate(NullBox, 1);
-		RegionSubtract(borderVisible,
-				      &pWin->borderClip, &pWin->winSize);
-		pWin->valdata->before.borderVisible = borderVisible;
-	    }
-	    pWin->valdata->before.resized = TRUE;
-	}
-    }
-
-    SetWinSize (pWin);
-    SetBorderSize (pWin);
-
-    ResizeChildrenWinSize(pWin, 0, 0, 0, 0);
-
-    if (WasViewable)
-    {
-	if (pWin->backStorage)
-	{
-	    pOldClip = RegionCreate(NullBox, 1);
-	    RegionCopy(pOldClip, &pWin->clipList);
-	}
-
-	anyMarked |= (*pScreen->MarkOverlappedWindows)(pWin, pWin,
-						(WindowPtr *)NULL);
-
-#ifdef DO_SAVE_UNDERS
-	if (DO_SAVE_UNDERS(pWin))
-	{
-	    dosave = (*pScreen->ChangeSaveUnder)(pLayerWin, pLayerWin);
-	}
-#endif /* DO_SAVE_UNDERS */
-
-	if (anyMarked)
-	    (*pScreen->ValidateTree)(pLayerWin->parent, NullWindow, VTOther);
-    }
-
-    if (pWin->backStorage &&
-	((pWin->backingStore == Always) || WasViewable))
-    {
-	if (!WasViewable)
-	    pOldClip = &pWin->clipList; /* a convenient empty region */
-	bsExposed = (*pScreen->TranslateBackingStore)
-			     (pWin, 0, 0, pOldClip,
-			      pWin->drawable.x, pWin->drawable.y);
-#ifdef NXAGENT_SERVER
-
-        /*
-         * We got a few, rare, segfaults here after having
-         * started using the backing store. It may be a
-         * different bug but miChangeSaveUnder() calls mi-
-         * CheckSubSaveUnder() that, in turn, can change
-         * the backing store attribute of the window. This
-         * means that we may try to destroy the region
-         * even if it was not created at the beginning of
-         * this function as, at the time, the backing store
-         * was off. miCheckSubSaveUnder() appear to get a
-         * pointer to the parent, so maybe doesn't change
-         * the attribute of the window itself. This is to
-         * be better investigated.
-         */
```

Comment:
Maybe we should also copy this comment in case we ever need to backtrack why `pOldClick != NULL` is being checked for?